pkg/harvester/integrity/repository/disk.go

```diff
@@ -1 +1,237 @@
 package repository
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	//"geninterface/repository"
+	"io/ioutil"
+	"math/big"
+	"net/url"
+	"time"
+
+	"github.com/andres-erbsen/clock"
+)
+
+// Disk is a SigningCA that generates signing certificates and validation materials
+// Disk is a local signing CA and has access to root keys and root certs via a path.
+type Disk struct {
+	rootSigner       crypto.Signer       // Root Key
+	rootCertificate  *x509.Certificate   // Root Certificate CA
+	validationBundle []*x509.Certificate // List of certificates to be used for validation of signing certificates
+}
+
+type Config struct {
+	CertFilePath           string `hcl:"cert_file_path"`           // Root CA certificate
+	KeyPath                string `hcl:"key_file_path"`            // Root Key
+	ValidationMaterialPath string `hcl:"validation_material_path"` // TrustBundle to validate CA signed material
```

Comment:
I don't know if I'm missing something but I don't know how exactly we would be using this validation material. I sketched this diagram to clarify how I'm seeing the chain of trust, see if that makes sense to you.

![image](https://user-images.githubusercontent.com/20044948/234725296-43750d5f-fbe0-481f-92b6-c2ca3a114df3.png)

This means that all we need to return from this layer is a single public key that was used to sign the certificates issued to harvesters. Am I missing something here?